src/utils.ts

```diff
@@ -97,7 +105,93 @@ export const getTracerInfo = (): { name: string; version: string } => {
   return { name, version };
 };
 
+const xRayFieldKeyValuePattern = /([^=;]+)=([^;]*)/g;
```

Comment:
As we discussed, we can simplify this into:
```js
const regex = /([^;=]+)=([^;=]+)/g;
const matches = Array.from(str.matchAll(regex)).slice(0, 5); // support up to 5 keys?
const result = Object.fromEntries(matches.map(match => [match[1], match[2]]));
```

Reply:
Done